### PR TITLE
feat: Problem description

### LINKED ISSUE #42: ユーザーは問題の解説を閲覧できる
## 完了条件

- 解説ボタンを押すとモーダルが表示される
- 問題ごと、言語ごとに表示する解説が異なる

## やらないこと

- 
- 
-

### PR DESCRIPTION
**Pull request by ykit00**
Close #42 

モーダルで問題の解説を表示できるようにしました。
image

## Self Check

- [x] I've confirmed `All checks have passed` on PR page. (You may leave this box unchecked due to long workflows.)
  - PR title follows [Angular's commit message format](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-commit-message-format).
    - PR title doesn't have `WIP:`.
  - All tests are passed.
    - Test command (e.g., `yarn test`) is passed.
    - Lint command (e.g., `yarn lint`) is passed.
- [x] I've reviewed my changes on PR's diff view.

### CODE REVIEW

File: src/app/(withAuth)/courses/[courseId]/programs/[programId]/CheckpointProblem.tsx [hunk 1]
```diff
@@ -90,7 +96,10 @@ export const CheckpointProblem: React.FC<CheckpointProblemProps> = ({
         </Box>
       </VStack>
       <VStack align="end" minW="50%" overflow="hidden">
-        <Button colorScheme="gray">解説</Button>
+        <Button colorScheme="gray" onClick={onOpen}>
```
Comment (Tatehito) [inline src/app/(withAuth)/courses/[courseId]/programs/[programId]/CheckpointProblem.tsx:99]: ExecutionResultProblem.tsx と同じように、CheckpointProblem.tsxやStepProblem.tsxにも `{explanation && ( ...` を入れた方が良いかも？

File: src/app/(withAuth)/courses/[courseId]/programs/[programId]/CheckpointProblem.tsx [hunk 2]
```diff
@@ -27,13 +30,16 @@ export const CheckpointProblem: React.FC<CheckpointProblemProps> = ({
...
+  programId,
   selectedLanguageId,
   setBeforeCheckPointLine,
   setCurrentCheckPointLine,
   setStep,
 }) => {
   const turtleGraphicsRef = useRef<TurtleGraphicsHandle>(null);
 
+  const { isOpen, onClose, onOpen } = useDisclosure();
+  const explanation = getExplanation(programId, selectedLanguageId);
```
Comment (Tatehito) [inline src/app/(withAuth)/courses/[courseId]/programs/[programId]/CheckpointProblem.tsx:42]: 些細なことですが、page.tsxで `const explanation = getExplanation(programId, selectedLanguageId);`  をして各コンポーネントに渡すようにすれば `getExplanation` の記述箇所が page.tsxだけで済むかもと思ったり？どちらが良いでしょうか...

Comment (ykit00) [reply src/app/(withAuth)/courses/[courseId]/programs/[programId]/CheckpointProblem.tsx]: たしかに親のほうでやったほうがよさそうですね！

**Review by Tatehito — APPROVED** [review]
LGTM!

**Pull request merged**